src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -1671,7 +1671,7 @@ BAD_RETURN(s32) func_80832224(Player* this) {
     this->unk_6AD = 0;
 }
 
-s32 func_8083224C(PlayState* play) {
+s32 Player_IsTalking(PlayState* play) {
     Player* this = GET_PLAYER(play);
 
     return CHECK_FLAG_ALL(this->actor.flags, ACTOR_FLAG_TALK);
```

Comment:
Player has two talk flags associated with it:
`this->actor.flags |= ACTOR_FLAG_TALK;`
and
`this->stateFlags1 |= PLAYER_STATE1_TALKING`
could you comment on the difference between these flags? `Player_IsTalking` is specifically checking that `ACTOR_FLAG_TALK` is set on player and not `PLAYER_STATE1_TALKING`, should this distinction be noted?

Reply:
The state flag is referring to links general state, while the actor flag is set specifically if link is given a text id from another actor to display. Technically it is possible to have `PLAYER_STATE1_TALKING` set and `ACTOR_FLAG_TALK` not set if the text id is -1. 
Link will appear to be in the talking state but no text will be on screen. I think its used in cutscenes but not too familiar on its use cases.

I dont really know what exactly to say on this, or where to put it

Reply:
What I said about textid -1 is only true for exchange items apparently lol. Its a little strange

Reply:
Basically I think that is everything was controlled with the state flag nothing would really be different, im not sure why the actor flag stuff even exists.
The only interesting thing about the actor flag is this block, where apparently if `Player_StartTalking` is called while player already has a text id set, it will be cleared to 0 
```c
        if (this->actor.flags & ACTOR_FLAG_TALK) {
            this->actor.textId = 0;
        } else {
            this->actor.flags |= ACTOR_FLAG_TALK;
            this->actor.textId = actor->textId;
        }
```
I have no idea if this ever gets used in practice though

Reply:
Thanks for the explanation, seems subtle enough to leave it as is then.